conda-incubator/conda-store
# DEV - Update dev workflows and CI

Split from #792 

## Description

- Remove OS-specific `environment.yaml` files as no longer needed
- Update env files - server
- Update environment file - core
- Add note about updating dependencies
- Update CI - remove unnecessary `yaml` files

## Pull request checklist




## Additional information


With this PR, at most, contributors/maintainers will need to regenerate/update their local conda environment. 

## How to test



💻 Locally:

1. Regenerate your local `conda` development environment (should use `conda-store-server/environment-dev.yaml`
2. Perform your usual development tasks such as running tests, building conda-store, etc.

To test the changes to the test GH workflows, you can use https://github.com/nektos/act and run the workflow or job accordingly. For example, to only run the conda-store integration tests: `act -j "integration-test-conda-store-server"`
